### PR TITLE
Foretagsanmalan design

### PR DESCRIPTION
**Pull request by Eggetxd**
råkade göra en pull request till stora hemsidan... hehe fixat nu

### CODE REVIEW

File: src/pages/f%C3%B6rf%C3%B6retag.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+                <p className="w-4/5 md:w-3/5">D-Dagen är Datasektionens årliga arbetsmarknadsdag. Det är ett heldagsevent där företag och studenter verksamma inom data och IT får en möjlighet att knyta kontakter och lära känna varandra. Detta skapar goda möjligheter för en mer direkt och personlig kontakt mellan företagsrepresentanter och studenter.</p>
+            </div>
+            <div className="text-center m-12">
+                <h2 className="text-cerise text-2xl mb-6 uppercase">ANMÄLAN TILL MÄSSAN</h2>
+                <p className="mb-8">BLAH BLAH BLAH BLAH</p>
+                <Link
+                    className="bg-cerise-strong p-2.5 rounded-full uppercase text-white"
+                    href="/företagsanmälan"
+                >Företagsanmälan</Link>
+            </div>
```
Comment (ernum) [inline src/pages/f%C3%B6rf%C3%B6retag.js:28]: This branch probably shouldn't include changes to the "for companies".

File: tailwind.config.js [hunk 2]
```diff
@@ -13,6 +13,10 @@ module.exports = {
           light: '#ec5f99',
           DEFAULT: '#e83d84',
         },
+      },
+      backgroundImage: {
+        'foretagBanner': "url('/img/foretag-banner.svg')",
+        'webBackground': "url('/img/website-background.svg')",
```
Comment (ernum) [inline tailwind.config.js:19]: This background isn't high-quality enough for the website. We should probably figure out how to tweak it to make it clearer.

File: src/components/CompanyForm.js [hunk 3]
```diff
@@ -25,82 +25,88 @@ export default function CompanyForm(props) {
...
                     <input 
+                        required
                         type="text" 
-                        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" 
+                        className="peer bg-transparent appearance-none border-b-2 border-gray-400 focus:border-cerise w-full py-2 px-1 text-gray-400 leading-tight focus:outline-none focus:shadow-outline" 
                         id="orgNum"
                         name="orgNum"
                         value={orgNum} 
                         onChange={(e) => {setOrgNum(e.target.value);}}
                     />
```
Comment (ernum) [inline src/components/CompanyForm.js:52]: Some inputs need the "[pattern](https://www.w3schools.com/tags/att_input_pattern.asp)" attribute to disallow companies from inputting the wrong format. We can use simple regexes for OrgNum, for example. Can you fix this in the pull request, or should we create a new one?

**Review by ernum — CHANGES_REQUESTED** [review]

File: src/pages/f%C3%B6rf%C3%B6retag.js (hunk 1 above)
Comment (Eggetxd) [reply src/pages/f%C3%B6rf%C3%B6retag.js:28]: yes, I forgot to change it from the last branch

File: tailwind.config.js (hunk 2 above)
Comment (Eggetxd) [reply tailwind.config.js:19]: I took it from figma so it is the highest quality I could get easily.

File: src/components/CompanyForm.js (hunk 3 above)
Comment (Eggetxd) [reply src/components/CompanyForm.js:52]: hmm I can take a look at that but I'll do that in a later pull request.

File: src/pages/f%C3%B6rf%C3%B6retag.js (hunk 1 above)
Comment (Eggetxd) [reply src/pages/f%C3%B6rf%C3%B6retag.js]: I have now removed the changes linked to the "for companies" page.

File: src/components/CompanyForm.js (hunk 3 above)
Comment (Eggetxd) [reply src/components/CompanyForm.js]: I have now changed the styling to support pattern which I have implemented for the email input. I am not sure of the format on the other inputs so I have not done them but it is as simple as writing: pattern="[regex]" in the desired input

Comment (mathiasmagnusson) [reply src/components/CompanyForm.js]: For emails, web browsers do this without having to specify a regex. Can be selected with e.g. `invalid:border-red-700`

File: src/components/CompanyForm.js [hunk 4]
```diff
@@ -25,82 +25,95 @@ export default function CompanyForm(props) {
...
                         value={invEmail} 
                         onChange={(e) => {setInvEmail(e.target.value);}}
                     />
+                    <label htmlFor="invEmail" className="transform transition-all absolute cursor-text text-gray-400 peer-focus:text-cerise font-bold text-sm peer-placeholder-shown:text-lg peer-focus:text-sm top-0 left-0 -translate-y-full peer-focus:-translate-y-full peer-placeholder-shown:translate-y-0 uppercase autofill:bg-transparent">{props.t.companyForm.invEmail}:</label>
                 </div>
 
-                <div className="flex items-center justify-between">
-                    <button className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline" onClick={(e) => addCompanyDocument(e)}>
+                <div className="flex flex-col items-center justify-between">
+                    <button className="bg-cerise hover:bg-cerise-light text-white font-bold py-2 px-4 rounded-full focus:outline-none focus:shadow-outline" onClick={(e) => addCompanyDocument(e)}>
```
Comment (mathiasmagnusson) [inline src/components/CompanyForm.js:116]: This should be an `input` with type `submit`, and the `onClick` here should instead be an `onSubmit` on the form

File: src/components/CompanyForm.js [hunk 5]
```diff
@@ -25,82 +25,95 @@ export default function CompanyForm(props) {
...
 
     return (
-        <div className="w-full max-w-lg ">
+        <div className="w-full flex justify-center py-16">
   
-            <form method="post" className="px-8 pt-6 pb-8 mb-4 bg-white rounded shadow-md ">
-                <div className="mb-4">
-                    <label htmlFor="name" className="block text-gray-700 font-bold mb-2 text-lg">{props.t.companyForm.name}:</label>
+            <form method="post" className="bg-transparent w-3/5">
+                <div className="my-16 relative">
```
Comment (mathiasmagnusson) [inline src/components/CompanyForm.js:31]: It would be nice to put this in a separate, non-exported function to avoid repetition

**Review by mathiasmagnusson — CHANGES_REQUESTED** [review]
Also make sure to rebase/merge this and make it a `.tsx` file

File: tailwind.config.js (hunk 2 above)
Comment (Eggetxd) [reply tailwind.config.js:18]: smart, I'll do that!

Comment (Eggetxd) [reply tailwind.config.js:18]: det tar MYCKET längre tid att ladda in den nu...

File: src/components/CompanyForm.js (hunk 3 above)
Comment (Eggetxd) [reply src/components/CompanyForm.js]: fixed

File: src/components/CompanyForm.js (hunk 4 above)
Comment (Eggetxd) [reply src/components/CompanyForm.js]: fixed!

**Review by mathiasmagnusson — APPROVED** [review]
The background is still problematic, but I think we can commit this until we have a better one.

**Comment by mathiasmagnusson** [comment]
Rebased onto `develop`

File: src/pages/f%C3%B6retagsanm%C3%A4lan.tsx [hunk 6]
```diff
@@ -1,13 +1,35 @@
...
+        <div className="mx-auto flex flex-col items-center text-center py-40">
+            <h1 className="md:max-w-2xl md:text-5xl mx-10 text-3xl text-cerise uppercase mb-24">{t.title}</h1>
+            <h2 className="text-lg text-white w-2/3 uppercase mb-5">{t.subtitle}</h2>
+            <p className="max-w-[350px] w-2/3 text-[#a7a7a7] mb-24">{t.text} <a href="mailto:ansvarig@ddagen.se" className="text-cerise hover:underline">ansvarig@ddagen.se</a></p>
+
+            <a href="mailto:ansvarig@ddagen.se" className="
+                bg-cerise transition-transform hover:scale-110
+                text-white font-bold uppercase
+                py-2 px-4 rounded-full
+            ">{t.contact}</a>
```
Comment (ernum) [inline src/pages/f%C3%B6retagsanm%C3%A4lan.tsx:16]: The padding on the y-axis should be less in my opinion. It's quite low.

image

File: src/components/CompanyForm.tsx [hunk 7]
```diff
@@ -2,110 +2,124 @@ import { type FormEvent, useState } from "react";
...
-            Företagsnamn: name,
-            Organisationsnummer: orgNum,
-            Kontaktperson: contact,
-            Meljadress: email,
-            Faktureringsadress: invAddress,
-            MejladressFörFakturering: invEmail
-        });
+        try {
+            setLoading(true);
+            await db.collection("company").add({
```
Comment (ernum) [inline src/components/CompanyForm.tsx:69]: This should be 'exhibitor'

**Review by ernum — CHANGES_REQUESTED** [review]

**Comment by ernum** [comment]
LGTM!

**Pull request merged**